### PR TITLE
ekf2: minimize fixCovarianceErrors() calls per update

### PR DESCRIPTION
**Pull request by dagar**
Needs careful review/testing.

### CODE REVIEW

File: src/modules/ekf2/EKF/ekf.cpp [hunk 1]
```diff
@@ -115,6 +115,14 @@ bool Ekf::update()
 
 		// run EKF-GSF yaw estimator
 		runYawEKFGSF();
+
+
+		// fix gross errors in the covariance matrix and ensure rows and
+		// columns for un-used states are zero
+		if (_covariance_updated) {
+			fixCovarianceErrors(true);
```
Comment (dagar) [inline src/modules/ekf2/EKF/ekf.cpp:123]: TODO: consider running this at the beginning of the next cycle instead? Then in the output data (estimator_status, etc) we might have some visibility in the necessity of these checks/fixes.

Comment (dagar) [reply src/modules/ekf2/EKF/ekf.cpp]: PR updated to test.

**Review by priseborough — COMMENTED** [review]
We can split the fix function into two. One will be to limit variances (thats the cheap one) and the other is to force symmetry (thats the expensive one). We can't afford to wait for a observation fusion event before limiting the maximum value of the sate variances or else if there is no observation for extended periods, the variances will continue to grow unchecked.

**Comment by priseborough** [comment]
There are three elements to the covariance health operations:

1) Variance (diagonals) lower limit. should be applied after every fusion step, ie when doing P = P-KHP
2) Variance (diagonals) upper limit should be applied every covariance prediction and fusion step.
3) Symmetry should be applied after every fusion step that is not a direct measurement of a state. Direct state measurements are things like NED position and NED velocity (GPS, Baro, etc). 

We may be able to relax item 3) to doing it prior to the start of the next prediction cycle (cheapest) or if that is not robust, then only at the end of the three fusion operations for a vector quantity, eg mag fusion, but this is untested.

**Pull request closed without merging**